analyzers/src/SonarAnalyzer.CSharp/Rules/ClassWithOnlyStaticMember.cs

```diff
@@ -79,18 +80,26 @@ private static void CheckConstructors(SonarSymbolReportingContext context, IName
                 return;
             }
 
-            var reportMessage = string.Format(MessageFormatConstructor, utilityClass.IsSealed ? SyntaxConstants.Private : SyntaxConstants.Protected);
-
             foreach (var constructor in utilityClass.GetMembers()
                                                     .Where(IsConstructor)
                                                     .Where(symbol => ProblematicConstructorAccessibility.Contains(symbol.DeclaredAccessibility)))
             {
                 var syntaxReferences = constructor.DeclaringSyntaxReferences;
                 foreach (var syntaxReference in syntaxReferences)
                 {
-                    if (syntaxReference.GetSyntax() is ConstructorDeclarationSyntax constructorDeclaration)
+                    switch (syntaxReference.GetSyntax())
                     {
-                        context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), reportMessage));
+                        case ConstructorDeclarationSyntax constructorDeclaration:
+                            var reportMessage = string.Format(MessageFormatConstructor, utilityClass.IsSealed ? SyntaxConstants.Private : SyntaxConstants.Protected);
+                            context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), reportMessage));
```

Comment:
Educational: isn't Diagnostic.Create() doing the `string.Format` already?
`context.ReportIssue(Diagnostic.Create(Rule, constructorDeclaration.Identifier.GetLocation(), MessageFormatConstructor, utilityClass.IsSealed ? SyntaxConstants.Private : SyntaxConstants.Protected));`